src/useDebounce.jsx

```diff
@@ -0,0 +1,15 @@
+import { useEffect } from "react";
+
+//кастомный хук можно его переиспользовать в любов участке кода
+const useDebounceEffect = (callback, delay, deps) => {
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      callback();
+    }, delay);
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [...deps, delay]);
+};
+
+export default useDebounceEffect;
```

Comment:
rework custom hook [useDebounce](https://dev.to/mukhammadsobirov/kastomnyi-react-khuk-usedebounce-2b3l)